modules/api/docs/LorisRESTAPI_v0.0.4-dev.md

```diff
@@ -823,90 +823,192 @@ The JSON object is of the form:
+field `File` with the imaging file, and `Json`, with the following keys and
+values:
+
+```js
+{
+    "CandID": $CandID,
+    "PSCID": $PSCID,
+    "VisitLabel": $VisitLabel,
+    "IsPhantom": boolean
+}
+```
+
+An optionnal header 'LORIS-Overwrite' with the value 'overwrite' can be used to
+overwrite an existing file.
+A successful request will be answered by a `303 See Other` response with its
+`Location` header pointing to the processes list of the new upload.
+(See 5.3 Tar Level processes)
+
 ### 5.2 Tar Level Data
+
+This section describe how to upload DICOM studies and how to start and monitor
+`mri_upload` processes.
+DICOM studies that have been successfully uploaded and processed can be downloaded with
+the following `GET` request:
+
 ```
-GET /candidates/$CandID/$VisitLabel/dicoms/$Tarname
+GET /candidates/$CandID/$VisitLabel/dicoms/$ArchiveName
 ```
 
-Returns/Downloads a `tar` file which contains a `.meta` and a `.log` text 
+Returns/Downloads a `tar` file which contains a `.meta` and a `.log` text
 files, and a `.tar.gz` of the raw DICOM data as acquired during the candidate
-scanning session, and as retrieved from `/candidates/$CandID/$Visit/dicoms`.
+scanning session, and as retrieved from `/candidates/$CandID/$VisitLabel/dicoms`.
 
-Only `GET` is currently supported.
+To get a list of the processes and their status for a given DICOM study previously uploaded use the following:
+
+```
+GET /candidates/$CandID/$VisitLabel/dicoms/$TarName/processes
+```
+
+The response contains all `mri_upload` attempts with the specified `$tarname`. And for
+each of them, a list of processes status.
+Response shape:
+
+```js
+{
+  "MriUploads": [
+    {
+      "MriUploadID": 123,
+      "Processes": [
+        {
+          "END_TIME": "YYYY-MM-DD hh:mm:ss",
+          "EXIT_CODE": "0",
+          "ID": "1",
+          "PID": "24971",
+          "PROGRESS": "text"
+          "STATE": "SUCCESS|RUNNING|ERROR"
+        },
+        ...
+      ]
+    },
+    ...
+  ]
+}
+```
+
+** An empty `Processes` array means that there has never been a process launched on
+that MRI upload.
+To start an MRI upload process on a previously uploaded DICOM study, a POST request
+containing the `MriUploadID` attribute in the request body should be sent.
+
+```
+POST /candidates/$CandID/$VisitLabel/dicoms/$TarName/processes
+```
+
+The request body must contain:
+
+```js
+{
+  "ProcessType": "mri_upload",
+  "MriUploadID": 123
+}
+```
+
+Expected response: 202 Accepted with `Location` header pointing to the new process.
+To obtain a specific process state, use the following:
+
+```
+GET /candidates/$CandID/$VisitLabel/dicoms/$TarName/processes/$ProcessID
+```
+
+Response shape:
+```js
+{
+  "ProcessState": [
+    {
+      "END_TIME": "YYYY-MM-DD hh:mm:ss",
+      "EXIT_CODE": "0",
+      "ID": "1",
+      "PID": "24971",
+      "PROGRESS": "text",
+      "STATE": "SUCCESS|RUNNING|ERROR"
+    }
+  ]
+}
+```
```

Comment:
Looks like the response for the get after a failed run is incorrect:
```
<Response [200]>
{'MriUploads': [{'MriUploadID': 124, 'Processes': [{'ID': 52, 'PID': 59709, 'STATE': 'SUCCESS', 'PROGRESS': 'MRI upload task (pid = 59709) failed: error code is 2', 'EXIT_CODE': 0, 'END_TIME': '2024-10-01 17:39:52'}]}]}
```

In that case, STATE should be failed and error code 2, no?

Reply:
Hhhhm, probably, this seems to be an error with the server process monitor not registering the correct exit code in the database (if that code is indeed 2). Not sure if it is related to this PR but I'll look into it.

Reply:
weird. OK. Sounds like a bug on the server process manager side then. Not sure why it is not getting the correct exit code... Maybe you could create an issue for this? Obviously, this will not get tackled in this PR.

Reply:
Is this thread resolved?

Reply:
Issue created (#9459). We agreed with Cécile that this is not an issue for this PR.